### DISCUSSION TITLE
Configure Custom claims

### DISCUSSION THREAD

**Discussion by satishviswanathan** [zitadel/zitadel#7611] [Q&A] [answered]
Zitadel version : 2.38.2

I was trying the following blog and was expecting the role to be added under custom claim "my:zitadel:grants". However when I inspect my id_token I couldn't find it. Not sure what I'm missing.  

https://zitadel.com/blog/custom-claims

Adding @dakshitha for any inputs.

**Comment by dakshitha** [comment]
Let me test this and get back to you.

**Reply by satishviswanathan** [reply]
Thank you appreciate it

**Reply by dakshitha** [reply, marked as answer]
I think there's a small typo in the blog post. The last `}` is absent and let's try `==` instead of `===`, as shown below. The following code works for me: 

```function flatRoles(ctx, api) {
  if (ctx.v1.user.grants == undefined || ctx.v1.user.grants.count == 0) {
    return;
  }

  let grants = [];
  ctx.v1.user.grants.grants.forEach(claim => {
    claim.roles.forEach(role => {
        grants.push(claim.projectId+':'+role)  
    })
  })

  api.v1.claims.setClaim('my:zitadel:grants', grants)
}
```

Thanks for pointing this out. Will make the fix in the blog post.

**Reply by satishviswanathan** [reply]
Thank you. That worked
